libnavigation-base/src/main/java/com/mapbox/navigation/base/utils/route/RouteUtils.kt

```diff
@@ -0,0 +1,162 @@
+package com.mapbox.navigation.base.utils.route
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteLeg
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.geojson.Point
+import com.mapbox.geojson.utils.PolylineUtils
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.base.utils.DecodeUtils.precision
+import com.mapbox.navigation.utils.internal.logE
+import com.mapbox.navigation.utils.internal.logW
+
+/**
+ * Route Utils
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+object RouteUtils {
+
+    private const val LOG_CATEGORY = "RouteUtils"
+
+    /**
+     * The method checks whenever the route has unexpected closures based
+     * on [RouteOptions.snappingIncludeClosuresList] and [RouteOptions.snappingIncludeStaticClosuresList]
+     * flags. If the any of them is **false** for the waypoint and closure overlaps it the method
+     * immediately returns **true**.
+     *
+     * If route has closures that are not overlapped waypoints, the method immediately returns
+     * **true**.
+     *
+     * Note: Experimental preview API
```

Comment:
```suggestion
     * This function checks whether the [navigationRoute] has unexpected closures, which could be a reason to re-route.
     *
     * `true` is returned whenever there's any [RouteLeg.closures] outside of a direct region around a coordinate from [RouteOptions.coordinatesList]
     * that was allowed for snapping to closures with [RouteOptions.snappingIncludeClosuresList] or [RouteOptions.snappingIncludeStaticClosuresList].
     * flags. Otherwise, `false` is returned.
```